tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/ScopedBuilder.scala

```diff
@@ -36,18 +37,23 @@ class ScopedBuilder
     extends BaseBuilder with BoolBuilder with ArithmeticBuilder with SetBuilder with SeqBuilder with ActionBuilder
     with FunBuilder with ControlBuilder with LiteralAndNameBuilder {
 
+  private def parser = DefaultType1Parser
+
   /**
    * Creates a `TBuilderInstruction` from a precomputed `TlaEx`. Voids correctness guarantees.
    *
    * Use sparingly, and only for expressions that have already passed static analysis.
    *
    * To use the builder outside of testing scenarios, where the expressions aren't written from scratch, it is necessary
-   * to be able to mark certain expressions as "trusted", e.g. when transforming and recombining invariants, or parts of
+   * to be able to use preexisting expressions, e.g. when transforming and recombining invariants, or parts of
    * Init/Next. While `build` is safe as a unidirectional implicit conversion from `TBuilderInstruction` to TlaEx, the
-   * inverse, `useTrustedEx`, needs to be explicit, to stress the fact that it should be used rarely, typically at most
+   * inverse, `unchecked`, needs to be explicit, to stress the fact that it should be used rarely, typically at most
    * once per transformation on the initial input.
    */
-  def useTrustedEx(ex: TlaEx): TBuilderInstruction = ex.point[TBuilderInternalState]
+  def unchecked(ex: TlaEx): TBuilderInstruction = ex.point[TBuilderInternalState]
+
+  /** Allows the use of type strings in the builder, to simplify writing complex types. */
+  def parseType(typeAsString: String): TlaType1 = parser.parseType(typeAsString)
```

Comment:
Why do we need this? It looks like it's only used once, in a test.

Reply:
Igor requested this:
https://github.com/informalsystems/apalache/pull/1772#discussion_r875619104